spring-test/src/main/java/org/springframework/test/web/client/response/ExecutingResponseCreator.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2002-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.test.web.client.response;
+
+import java.io.IOException;
+
+import org.springframework.http.client.ClientHttpRequest;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.mock.http.client.MockClientHttpRequest;
+import org.springframework.test.web.client.ResponseCreator;
+import org.springframework.util.Assert;
+import org.springframework.util.StreamUtils;
+
+/**
+ * A {@code ResponseCreator} which delegates to a {@link ClientHttpRequestFactory}
+ * to perform the request and return the associated response.
```

Comment:
I think it would be good to have a second sentence that starts with "This is useful when...". As far as I can tell it's about testing code that calls multiple remote service, where some need should be called rather than mocked.

Reply:
thanks for the feedback. I've addressed the documentation and discoverability suggestions.